t/alignment/algx.xml

```diff
@@ -31,7 +31,7 @@
       <listingline xml:id="algx1.l2"><tags>
           <tag><text fontsize="80%">2:</text></tag>
           <tag role="refnum">2</tag>
-        </tags>     <text font="bold">return</text> 7
+        </tags>  <text font="bold">return</text> 7
```

Comment:
The leading space reduction in the algx test, and the algorithmic snippet in the [figure_mixed_content](https://github.com/brucemiller/LaTeXML/blob/2bef09e696a4b8d5b77f1ce38badce95c9165dca/t/complex/figure_mixed_content.tex#L264-L288) test rang a bell, where we've had two arXiv reports mentioning the leading whitespace in listings is getting wrongly removed. That's [arXiv/html_feedback 1712](https://github.com/arXiv/html_feedback/issues/1712) and [arXiv/html_feedback 2201](https://github.com/arXiv/html_feedback/issues/2201) for reference.

Maybe the underlying trickiness comes from even trying to model whitespace as actual characters inside an XML document (rather than some `<mspace>`-like element with a spacing value). But since we're doing characters, I wonder if it's worth taking a closer look at what exactly changed in those `{algorithmic}`-related tests, and if it is the same cause as the `{lslistings}` reports.

Reply:
I now understand that the line I am commenting on had 6 ascii spaces (U+0020) converted to two alternative space chars: (U+2003)(U+2002)

I should probably take some time next to study how successful various fonts are in getting the "expected" widths displayed. GitHub's review interface seems to be unsuccessful?

Reply:
The trigger for this PR was noticing that some kerns & skips were being dropped (they show up a *lot* in plain tex stuff); followed by improving the accuracy of converting them to spaces and filling in a few unimplemented commands.
That said, I too was surprised at being reminded that these spaces are (still) actually relied on for layout in listings. Even so, maybe that's enough to fix the cases you mention; I checked the 2nd one and the algorithms now come out indented nicely.
It would seem better in a *few* cases (not all kerns & skips!) to use an element w/width and CSS, rather than just clever spaces. The constructors for `\hskip,\kern` could do that, if it could detect the situation.  But I'm not sure how to define it; in this case possibly following an `ltx:tags` would be sufficient, but I haven't thought it through.